shared-bindings/keypad/KeyMatrix.c

```diff
@@ -82,20 +83,24 @@
 //|           maximum number of key transition events that are saved.
 //|           Must be >= 1.
 //|           If a new event arrives when the queue is full, the oldest event is discarded.
+//|         :param int debounce_threshold: Emit events for state changes only after a key has been
+//|           in the respective state for ``debounce_threshold`` scans on average.
+//|           The default is 1 (no debouncing), the maximum is 127.
```

Comment:
As we discussed earlier and I think you confirmed [here](https://github.com/adafruit/circuitpython/issues/8777#issuecomment-1913689607), I think a "scan" here means elapsed time of one `interval`. Is that indeed what you mean? So is `1` the same as the previous algorithm? I would think that 0 would mean no debouncing then.

I would change `scans on average` to `intervals` on average, assuming that's what you mean. It's not so obvious that a scan occurs once per interval, and in the future, it might even be true that it's implemented in a different way. 

If that is the case, would `debounce_intervals` be a better name?